### ISSUE TITLE
Issues with devices in gradients

### ISSUE THREAD

**Issue by Linux-cpp-lisp** [intel/intel-extension-for-pytorch#181]
When running on `ipex.DEVICE` instead of `cpu`, my model throws:
```
RuntimeError: Function MulBackward0 returned an invalid gradient at index 0 - expected type TensorOptions(dtype=float, device=cpu, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt)) but got TensorOptions(dtype=float, device=xpu:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
```
Using gradient anomaly mode shows that this occurs in the first operation before a `scatter` operation from `torch_scatter`, which may the root of the issue.

So far I am unable to reproduce the issue with a simple function outside of the model. I will update this issue if I am able to find a small reproducing example, but wanted to post to see if errors like this are familiar to the developers and if they have any suggestions for what the cause may be.

Thanks!

**Comment by EikanWang** [comment]
May I know your model?

**Comment by Linux-cpp-lisp** [comment]
Unfortunately I can't share the model itself right now, but I will let you know if I can reproduce it with a minimal example

**Comment by EikanWang** [comment]
I suppose this issue has been fixed on the release/1.10 branch. Please feel free to reopen it if not.

**Comment by Linux-cpp-lisp** [comment]
Yep, this issue no longer occurs on the latest version--- thanks!
